lib/CodeGen/PagerandoBinning.cpp

```diff
@@ -197,8 +229,14 @@ void SimpleBinner::getAnalysisUsage(AnalysisUsage &AU) const {
   PagerandoBinnerBase::getAnalysisUsage(AU);
 }
 
-
 PagerandoBinnerBase::Bin FirstFitAlgo::assignToBin(unsigned FnSize) {
+  if (MaxBinSize > 0) {
+    if (isUniPOT && (BinCount > MaxBinSize)) {
+        report_fatal_error(
+          "Pagerando w/ Unified POT is active, but the POT is exceeding a single page. Please increase bin-size with `-pagerando-bin-size=N`");
```

Comment:
Just a couple of nitpicks: "is exceeding" -> "exceeds", "bin-size" -> "bin size"